File: GFG/Largest subtree sum in a tree/Largest subtree sum in a tree.java
class Solution {
    static int ans;
    public static int findLargestSubtreeSum(Node root) {
        // code here
        ans=Integer.MIN_VALUE;
        dfs(root);
        return ans;
    }
    static int dfs(Node root){
        int sum=root.data;
        if(root.left!=null){
            sum+=dfs(root.left);
        }
        if(root.right!=null){
            sum+=dfs(root.right);
        }
        ans=Math.max(ans,sum);
        return sum;
    }
}

File: GFG/Largest subtree sum in a tree/Largest subtree sum in a tree.cpp
class Solution {
  public:
    int subtree(Node* root, int& ans){
        if(root==NULL){
            return 0;
        }
        int lsum = subtree(root->left,ans);
        int rsum = subtree(root->right,ans);
        int tot = root->data + lsum + rsum;
        ans = max(ans,tot);
        return tot;
    }
    int findLargestSubtreeSum(Node* root){
        int ans = INT_MIN;
        subtree(root,ans);
        return ans;
        
    }
};
